lib/Listener/FollowUpClassifierListener.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2024 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Listener;
+
+use DateInterval;
+use DateTimeImmutable;
+use OCA\Mail\BackgroundJob\FollowUpClassifierJob;
+use OCA\Mail\Db\ThreadMapper;
+use OCA\Mail\Events\NewMessagesSynchronized;
+use OCP\BackgroundJob\IJobList;
+use OCP\DB\Exception;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @template-implements IEventListener<Event|NewMessagesSynchronized>
+ */
+class FollowUpClassifierListener implements IEventListener {
+
+	public function __construct(
+		private LoggerInterface $logger,
+		private ThreadMapper $threadMapper,
+		private IJobList $jobList,
+	) {
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof NewMessagesSynchronized)) {
```

Comment:
Messages are classified when they are first seen. Assuming reliable background sync you will often first see the message added to the cache, then a reply happens, if any.
If we delay this check by 5-6 days we can skip classification of emails that have received a reply by then.

Reply:
That is a good point. It would certainly save a lot of processing power.